components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -330,6 +350,7 @@ func convertWorkspaceInfo(input *protocol.WorkspaceInfo) (*v1.Workspace, error)
 		Status: &v1.WorkspaceStatus{
 			Instance: instance,
 		},
+		Pinned: input.Workspace.Pinned,
```

Comment:
Note that Workspace listing can be done by other people as well (not just the workspace owner). For them, the Pinned property is actually incorrect, as the Pinned happens to be per user right now.